### PR TITLE
fix(dice): engine correctness gaps + DiceTestbench

### PR DESCRIPTION
**Pull request by Fronix**
## Summary

Fixes four correctness gaps in `DamageRoll` (advantage tie-break, multi-source adv/dis aggregation, AoE auto-flagging, no-proficiency crit suppression) and adds a developer dice testbench gated behind a new `Debug Mode` setting for verifying engine behavior interactively.

> **Stacked PR series** — this is PR 1 of 5 splitting the dice engine work from `feature/nimble-dice-update` into reviewable chunks:
> 1. **This PR** — Pipeline refactor + DiceTestbench (24 commits, 745 tests)
> 2. `dice/stage-1` — Decouple `explosionStyle` from `canCrit`
> 3. `dice/stage-2` — Die modifier vocabulary (`c`/`cv`/`v`/`n`)
> 4. `dice/stage-3` — `critCount`, `brutalPrimary`, `primaryDieValue` API
> 5. `dice/stage-4` — Post-roll value mutation framework
>
> Supersedes #573 (same scope, narrower branch).

## Changes

- **Bug fixes in `DamageRoll`:**
  - Leftmost-on-tie discarding now guaranteed via custom `khn`/`kln` Die modifiers registered at init.
  - Multiple adv/dis sources aggregate via `rollModeSources: number[]` — cancel 1-for-1 instead of last-writer-wins.
  - AoE damage nodes auto-set `canCrit: false, canMiss: false` from `activation.template.shape`.
  - New `weaponType` field on `ObjectDataModel` + `hasWeaponProficiency` helper for no-proficiency crit suppression. Migration015 backfills.
- **`DamageRoll` internals:**
  - Outcome interpretation lifted into `_finalizeOutcome` for clarity.
  - `_applyPostRollMutations` no-op extension point wired between rolling and finalization.
  - Vicious chain rerolls tagged with `provenance` for visualizers.
- **Developer testbench (`src/view/debug/`):**
  - `Debug Mode` client setting with "Open Dice Testbench" button.
  - Headless `stageAndRoll` orchestration with deterministic RNG + DSN compat.
  - Svelte 5 testbench UI: scenario presets, roll builder, categorized results panel.
  - 18 scenario presets covering baselines, crit, miss, flag suppression, and edge cases.
- **Tests:** 77 in `DamageRoll.test.ts` + 10 in `stageAndRoll.test.ts` = 745 total green.
- **Docs:** `docs/system/dice-engine.md` plain-English engine overview.

## Test Plan

- [x] `pnpm check` passes (format, lint, circular deps, type-check, 745 tests)
- [ ] Manual: enable Debug Mode → Open Dice Testbench → run Vicious d12 Force Crit, AoE cone, Minion scenarios
- [ ] Manual: verify normal weapon attacks from character sheets still work

## Checklist

- [x] Added or updated unit tests
- [x] PR targets the `dev` branch
- [x] `pnpm check` passes
- [x] No hardcoded user-facing strings (used `localize()`)
- [x] **Have you used AI to assist with this PR?** yes
- [x] **If yes:** reviewed against STYLE_GUIDE.md and AGENTS.md

### CODE REVIEW

File: docs/system/dice-engine.md [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+
+A plain-English guide to how the Nimble dice engine works in this system. Written for teammates, content authors, and future-you — not for someone doing surgery on the engine internals.
+
+## What makes Nimble dice weird
+
+Most TTRPGs roll dice and add things up. Nimble does that too, but with a twist that breaks how most dice engines are built:
+
+**Only one die — the "Primary Die" — decides whether you hit, miss, or crit.** The rest of the dice in your damage pool just add to the total.
+
+The Primary Die is whichever base-weapon die is **leftmost** in your pool. Everything else (sneak attack, elemental riders, Vicious extras, +X damage from a buff) adds to damage but **can never** cause a crit or a miss.
```
Comment (trevlar) [inline docs/system/dice-engine.md]: I think that Elemental Riders is a hallucination or a mistranscription. Should it actually mean elemental surges?

File: src/view/debug/DiceTestbench.svelte [hunk 2]
```diff
@@ -0,0 +1,1181 @@
...
+			} else {
+				// Leave a hole — but stageAndRoll has no hole concept, so just stop staging
+				// once we hit a blank (subsequent dice roll real). Use a sentinel: break.
+				break;
+			}
+		}
+		await performRoll(staged);
+	}
+
+	const resultJson = $derived(lastResult ? JSON.stringify(lastResult, null, 2) : '');
```
Comment (trevlar) [inline src/view/debug/DiceTestbench.svelte]: I think these 364 lines in the script tag would be good to split into separate files.

A *.d.ts file to contain and export the interface and types, a file to hold the constants, a separate file to hold state in src/view/debug/DiceTestbench.svelte.ts, and a separate file for utility files.

File: docs/system/dice-engine.md [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+Content authors can leave `weaponType` blank (nothing breaks) or set it for weapons that require proficiency. Migration014 backfilled existing weapons to `''` so nothing in existing worlds silently stopped critting.
+
+### Advantage/disadvantage sources
+
+`ItemActivationManager.activate()` accepts an optional `rollModeSources: number[]` in addition to the old single-integer `rollMode`. Features that grant or impose adv/dis should push an entry onto this array — everything gets netted before the roll. Backward compatible: if a caller still passes just a single integer, that works too.
+
+## What the engine does NOT do
+
+- **It doesn't render the chat card.** That lives in `src/view/chat/components/` and consumes the finalized roll.
+- **It doesn't integrate with Dice So Nice.** DSN visualizes whatever dice are on the roll; the engine has already decided the outcome before DSN sees anything. One current quirk: the vicious explosion path manually rerolls the primary die to avoid DSN preempting the crit chain. Don't touch that without talking to whoever owns DSN integration.
```
Comment (trevlar) [inline docs/system/dice-engine.md:104]: I thought this issue was resolved. Did we go backward on it?
We don't need to roll the primary die beforehand. Maybe a future PR of yours addresses this. The vicious roll is intended to be rolled with the explosion and will have a left most dice of the two that determine whether the dice crits again or not.

**Review by trevlar — APPROVED** [review]

File: src/view/debug/DiceTestbench.svelte (hunk 2 above)
Comment (Fronix) [reply src/view/debug/DiceTestbench.svelte]: I agree

File: docs/system/dice-engine.md (hunk 1 above)
Comment (Fronix) [reply docs/system/dice-engine.md]: This was just some jargon that had been added, it meant "bonus elemental damage" in this case.

File: docs/system/dice-engine.md (hunk 3 above)
Comment (Fronix) [reply docs/system/dice-engine.md:104]: Yes it has been resolved, i'll adjust the text in a later PR to clarify.
